core/src/main/java/com/datastax/oss/driver/internal/core/cql/CqlPrepareAsyncProcessor.java

```diff
@@ -53,23 +58,73 @@
 public class CqlPrepareAsyncProcessor
     implements RequestProcessor<PrepareRequest, CompletionStage<PreparedStatement>> {
 
+  public class CacheEntry {
+
+    private CompletableFuture<PreparedStatement> result;
+    private Set<CompletableFuture<PreparedStatement>> futures;
+    private AtomicBoolean lock;
+
+    public CacheEntry() {
+
+      result = new CompletableFuture<>();
+      futures = Sets.newHashSet(result);
+      lock = new AtomicBoolean(false);
+    }
```

Comment:
Small nit: could probably simplify this by initializing all of these in the variable declaration and then getting rid of the constructor.  Should also make these fields all final.